packages/integrations/cloudflare/src/index.ts

```diff
@@ -90,33 +94,91 @@ export default function createIntegration(args?: Options): AstroIntegration {
 					vite.ssr.target = 'webworker';
 				}
 			},
+			// BUG: `entrypoints` currently does not respect SERVER_BUILD_FOLDER, aka `build.server`
+			'astro:build:ssr': ({ manifest, entryPoints }) => {
+				// NOTE: monkey-patch because of entrypoints bug:
```

Comment:
Which bug?

Reply:
@alexanderniebuhr this is not a bug. The entry points, **in that hook**,  are correct.

I suppose you want to have `entryPoints` in the hook `astro:build:done`

Reply:
> @alexanderniebuhr this is not a bug. The entry points, **in that hook**, are correct.

I am quite confident that the paths returned in entryPoints are not correct. Please see https://github.com/withastro/astro/pull/7464#discussion_r1242273789 for example.

It's fine to have them in `astro:build:done`, it would also be fine to keep the `astro:buiod:ssr` step.

Reply:
> I am quite confident that the paths returned in entryPoints are not correct

They **are correct** because they are the physical paths returned by Rollup, which is responsible for generating the files.

The issue is that later, during the build step, `astro` moves those files in `build.server` folder. I am working on adding the correct data in a proper hook.